monai/apps/auto3dseg/data_analyzer.py

```diff
@@ -259,17 +267,29 @@ def get_all_case_stats(self, key="training", transform_list=None):
                         DataStatsKeys.LABEL_STATS: d[DataStatsKeys.LABEL_STATS],
                     }
                 )
-            result[DataStatsKeys.BY_CASE].append(stats_by_cases)
+            result_bycase[DataStatsKeys.BY_CASE].append(stats_by_cases)
+
+        n_cases = len(result_bycase[DataStatsKeys.BY_CASE])
 
-        result[DataStatsKeys.SUMMARY] = summarizer.summarize(cast(list, result[DataStatsKeys.BY_CASE]))
+        result[DataStatsKeys.SUMMARY] = summarizer.summarize(cast(list, result_bycase[DataStatsKeys.BY_CASE]))
+        result[DataStatsKeys.SUMMARY]["n_cases"] = n_cases
+        result[DataStatsKeys.BY_CASE] = [None] * n_cases
 
         if not self._check_data_uniformity([ImageStatsKeys.SPACING], result):
             print("Data spacing is not completely uniform. MONAI transforms may provide unexpected result")
 
         if self.output_path:
+            # saving summary and by_case as 2 files, to minimize loading time when only the summary is necessary
             ConfigParser.export_config_file(
```

Comment:
Hi @holgerroth, this export_config_file may have some efficiency issue, will this affect the fl stats implementation as well?

Reply:
Do we need to update the output path [here](https://github.com/Project-MONAI/MONAI/blob/a8302eca3fc31554d4bd30fae84150766ce13472/monai/fl/client/monai_algo.py#L219) in the FL client?

Reply:
that looks fine to me, this PR will affect modules that consume `data_stats.yaml`, not generate `data_stats.yaml`

Reply:
the FL related test still doesn't work, https://github.com/Project-MONAI/MONAI/actions/runs/4410200974/jobs/7727413151 do you have time to fix it? otherwise I can try to update this PR.